File: 5-Problem-A-Day/82. Remove Duplicates from Sorted List II.java
/**
 * Definition for singly-linked list. public class ListNode { int val; ListNode next; ListNode(int
 * x) { val = x; } }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null)
            return null;
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode prev = dummy;
        int count = 1;
        while (head.next != null) {
            if (head.val != head.next.val) {
                if (count == 1) {
                    prev.next = head;
                    prev = head;
                }
                count = 1;
            } else
                count += 1;
            head = head.next;
        }
        if (count == 1) {
            prev.next = head;
            prev = head;
        }
        prev.next = null;
        return dummy.next;
    }
}

File: 2024/82. Remove Duplicates from Sorted List II.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* dummy = new ListNode(101), *prev = dummy;
        while (head != NULL) {
            ListNode* curr = head;
            int val = head->val, count = 0;
            while (head != NULL && head->val == val) {
                count += 1;
                head = head->next;
            }
            if (count == 1) {
                prev->next = curr;
                prev = curr;
                prev->next = NULL;
            }
        }
        return dummy->next;
    }
};